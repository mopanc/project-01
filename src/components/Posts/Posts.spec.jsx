import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'Title 1',
      body: 'Body 1',
      cover: 'img/img1.png',
    },
    {
      id: 2,
      title: 'Title 2',
      body: 'Body 2',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: 'Title 3',
      body: 'Body 3',
      cover: 'img/img3.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);

    expect(screen.getAllByRole('heading', { name: /Title/i })).toHaveLength(3);
    expect(screen.getAllByRole('img', { name: /Title/i })).toHaveLength(3);
    expect(screen.getAllByText(/body/i)).toHaveLength(3);
    expect(screen.getByRole('img', { name: /Title 3/i })).toHaveAttribute('src', 'img/img3.png');
  });

  it('should not render posts', () => {
    render(<Posts />);
    expect(screen.queryByRole('heading', { name: /Title/i })).not.toBeInTheDocument();
  });

  it('should mach snapshot', () => {
    const { container } = render(<Posts {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
