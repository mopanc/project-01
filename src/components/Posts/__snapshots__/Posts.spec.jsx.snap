// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should mach snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Title 1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 1
         
      </h1>
      <p>
        Body 1
      </p>
      <span>
        1
      </span>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 2
         
      </h1>
      <p>
        Body 2
      </p>
      <span>
        2
      </span>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Title 3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        Title 3
         
      </h1>
      <p>
        Body 3
      </p>
      <span>
        3
      </span>
    </div>
  </div>
</div>
`;
